simplq/src/components/pages/Join/index.jsx

```diff
@@ -14,28 +15,33 @@ import InputField from '../../common/InputField';
 
 export function JoinQueueWithDetails(props) {
   const queueId = props.match.params.queueId;
-  const queueName = useSelector((state) => state.appReducer.queueName);
+  const [queueStatusResponse, setQueueStatusResponse] = useState();
+  useEffect(
+    () => QueueService.getStatus(queueId).then(setQueueStatusResponse).catch(handleApiErrors),
+    [queueId]
```

Comment:
Does setQueueStatusResponse get called with the status argument here automatically? I didn't realize this syntax worked.

Reply:
ha... this is cool... when you have a lambda foo,
```
const foo = (a) => console.log("I'm foo, ", a);
```

and you want to use it in a then, you can do:
```
.then(resp => foo(resp));
```

but then it's the same effect as,
```
.then(foo);
```

Then takes a lambda as calls it with the response.

Reply:
Good to know.

Reply:
This reminds me of method references in Java streams